### ISSUE TITLE
[BottomNavigation] UITabBarItem.accessibilityHint not assigned to item view

### ISSUE THREAD

**Issue by romoore** [material-components/material-components-ios#7290]
The `accessibilityHint` of the UITabBarItem is not being correctly passed through to the (button in) the item view. Users should be able to set the `accessibilityHint` of the item and VoiceOver should read it out.  Similarly, the value should be Key-Value Observed so changes are propagated to the item even after assignment.

---
#### Internal data
- Associated internal bug: [b/132166866](http://b/132166866)
